.github/workflows/ci.yml

```diff
@@ -0,0 +1,14 @@
+# for now
+on: push
+
+jobs:
+    check:
+        name: cargo test
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+            - uses: dtolnay/rust-toolchain@nightly
```

Comment:
After a5e33cc6473b51ce30cb1dfb01c571fef28f7da2 the code needs nightly rust to build/test, because C varadic support is still marked as a nightly feature and it's used in the libretro log interface implementation.

I had debated making the log interface enabled based on a cargo feature but idk how much worth it would be.

Reply:
For experimentation I did actually end up trying my second path, because I thought it would probably get to a working state faster (or, slightly more pessimistic, at all), and probably unsurprisingly, putting the formatting machinery in C++ that calls back to Rust works just fine. The only build-dep it should add is the cc crate like mentioned; bindgen and any other heavy machinery like that aren't needed because writing the definitions for the helper (and reciever) are simple (i.e: one function)  enough to do by hand.

I'm not that enthralled about having to do this for +stable compatibility 
(which is more a jab at the C varadic RFC/implementation being unstable for like 8 years..) but at least it works I guess :shrug: 

I'll push it and if it's too crispy I guess it can be reverted,,